server/src/main/java/org/elasticsearch/index/snapshots/blobstore/BlobStoreIndexShardSnapshots.java

```diff
@@ -250,15 +259,19 @@ public static BlobStoreIndexShardSnapshots fromXContent(XContentParser parser) t
                         while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
                             if (token == XContentParser.Token.FIELD_NAME) {
                                 currentFieldName = parser.currentName();
-                                if (parser.nextToken() == XContentParser.Token.START_ARRAY) {
-                                    if (ParseFields.FILES.match(currentFieldName, parser.getDeprecationHandler()) == false) {
-                                        throw new ElasticsearchParseException("unknown array [{}]", currentFieldName);
-                                    }
+                                if (ParseFields.FILES.match(currentFieldName, parser.getDeprecationHandler()) &&
```

Comment:
Funny enough with the strange way the serialization worked here, we actually didn't need any BwC logic to keep it readable by older ES versions, that's why there's no version specific serialization logic here.

Reply:
Should we make the parsing non-lenient now so that we are not silently dropping fields in the future?

Reply:
Yea let's do that but maybe keep that in a separate PR? The parsing here is pretty weird and overly complicated anyway IMO, I'd clean it up and make it non-lenient in one clean go after adding this change?